File: code/backend/php/src/Core/Domain/Story/DifficultyInfo.php
<?php

declare(strict_types=1);

namespace Games\Paths\Core\Domain\Story;

class DifficultyInfo
{
    public function __construct(
        public string $uuid,
        public ?string $description = null,
        public int $expCost = 5,
        public int $maxWeight = 10,
        public int $minCharacter = 1,
        public int $maxCharacter = 4,
        public int $costHelpComa = 3,
        public int $costMaxCharacteristics = 3,
        public int $numberMaxFreeAction = 1,
        public ?int $idCard = null,
        public ?CardInfo $card = null,
        public int $life = 100,
        public int $energy = 100,
        public int $sad = 0,
        public int $dexterity = 10,
        public int $intelligence = 10,
        public int $constitution = 10,
        public int $weight = 10,
        // Step 23 — trait cost budgets; null = no limit
        public ?int $traitCostPositiveBudget = null,
        public ?int $traitCostNegativeBudget = null
    ) {
    }
}

File: code/backend/python/app/core/models/story/difficulty_info.py
from dataclasses import dataclass, field
from typing import List
from app.core.models.story.card_info import CardInfo

@dataclass
class DifficultyInfo:
    uuid: str
    description: str | None = None
    expCost: int = 5
    maxWeight: int = 10
    minCharacter: int = 1
    maxCharacter: int = 4
    costHelpComa: int = 3
    costMaxCharacteristics: int = 3
    numberMaxFreeAction: int = 1
    idCard: int | None = None
    card: CardInfo | None = None
    life: int = 100
    energy: int = 100
    sad: int = 0
    dexterity: int = 10
    intelligence: int = 10
    constitution: int = 10
    weight: int = 10
